src/mbgl/tile/geometry_tile_worker.cpp

```diff
@@ -280,7 +282,7 @@ bool GeometryTileWorker::hasPendingSymbolDependencies() const {
 }
 
 void GeometryTileWorker::attemptPlacement() {
-    if (!data || !layers || !placementConfig) {
+    if (!placementConfig) {
```

Comment:
We don't need access to `data` and `layers` anymore, since all of that is encoded in the `SymbolLayout` objects now. This change is necessary, since disengaging the `optional` will make this check fail when it is part of `redoLayout()` (called at the very end), meaning that there won't be any symbols until placement is triggered again.